viewing_party/party.py

```diff
@@ -1,23 +1,184 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie_dict['title'] = title
+        movie_dict['genre'] = genre
+        movie_dict['rating'] = rating
+    
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)

```

Comment:
So now that we have talked about Big O analysis, we know that `.remove` has a time complexity of `O(n)` since it needs to look through every element in a list to find the sought after value. Right now, this loop has an outer loop of  `O(n)` and you when you find the movie to remove you have another `O(n)` operation. How could we circumvent using `remove`? (Hint: Think about list comprehension) 

Also in this implementation, we will only use `remove` once, but in other filtering problems we might have to do this for multiple elements so it doesn't hurt to get in the habit now.